### ISSUE TITLE
rewrite plugin doesn't appear to work

### ISSUE THREAD

**Issue by cognitiaclaeves** [fluent/fluent-plugin-rewrite-tag-filter#68]
https://groups.google.com/forum/#!topic/fluentd/lLi5ITFE5bE

Trying to figure out if I'm using the rewrite filter correctly. The following doesn't work:

```
<source>
  @type dummy
  dummy {"month":"05","day":"16","year":"2018","hour":"11","minute":"39","second":"04","milli":"175","msg":"*** DUMMY *** SQLWWT\twork done, returning to upstream thread","logPath":"/alloc/logs/sql1.log.05-16-2018.log"}
  tag rfa_wedge2.stg_dev.wedge.core.alloc.logs.sql1.log.05-16-2018.log
  auto_increment_key seq
</source>

<source>
  @type dummy
  dummy {"month":"05","day":"16","year":"2018","hour":"11","minute":"38","second":"01","milli":"913","msg":"*** DUMMY *** SQLWWT\twork done, returning to upstream thread","logPath":"/alloc/logs/sql1.log.05-16-2018.log","poi":"","path":"debug.sql1","@timestamp":"2018-05-16T11:38:01.913-0000","hostname":"wedge-0.rfa-dev-q2w-login","program":"rfa-dev-q2w-login","app":"wedge","requestId":"","message":"*** DUMMY *** RFA rfa.wedge.debug RFA SQLWWT\twork done, returning to upstream thread"}
  tag rfa_wedge2.stg_dev.rfa.wedge.debug.alloc.logs.sql1.log.05-16-2018.log
  auto_increment_key seq
</source>

<match rfa_wedge2.stg_dev.wedge.core.**>
  @type rewrite_tag_filter
  <rule>
    key     logPath
    pattern /.*sql.*/
    tag     rfa_wedge2.stg_dev.skip.${tag}
  </rule>
</match>

<match *.**>
  @type stdout
</match>
```

```
# 2018-05-16 21:14:10 +0000 [info]: starting fluentd-1.0.2 pid=9 ruby="2.4.2"
# 2018-05-16 21:14:10 +0000 [info]: spawn command to main:  cmdline=["/opt/td-agent/embedded/bin/ruby", "-Eascii-8bit:ascii-8bit", "/usr/sbin/td-agent", "--under-supervisor"]
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-elasticsearch' version '2.4.0'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-kafka' version '0.6.5'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-record-modifier' version '1.0.2'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-rewrite-tag-filter' version '2.0.1'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-s3' version '1.1.0'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-td' version '1.0.0'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-td-monitoring' version '0.2.3'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluent-plugin-webhdfs' version '1.2.2'
# 2018-05-16 21:14:10 +0000 [info]: gem 'fluentd' version '1.0.2'
```

I would expect these tags to be emitted:

rfa_wedge2.stg_dev.skip.rfa_wedge2.stg_dev.wedge.core.alloc.logs.sql1.log.05-16-2018.log
rfa_wedge2.stg_dev.rfa.wedge.debug.alloc.logs.sql1.log.05-16-2018.log

But I'm never seeing a rfa_wedge2.stg_dev.skip.** tag.

Do I need to specify a ^ at the front of the match string?

**Comment by okkez** [comment]
fluent-plugin-rewrite-tag-filter v2.0.1 does not support `pattern /.*sql.*/`, you can use `pattern .*sql.*` instead.

**Comment by okkez** [comment]
Released v2.1.0.
It supports patterns with slashes, try it, please.
